config_registry.go

```diff
@@ -0,0 +1,82 @@
+package evergreen
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/mongodb/grip"
+)
+
+var registry *configSectionRegistry
```

Comment:
can we name this configRegistry

(you can make more friendly accessor functions)